src/VisualStudio/Core/Def/Implementation/LanguageClient/InProcLanguageServer.cs

```diff
@@ -52,11 +53,27 @@ internal class InProcLanguageServer
         [JsonRpcMethod(Methods.InitializeName)]
         public Task<InitializeResult> Initialize(JToken input, CancellationToken cancellationToken)
         {
+            // The VS LSP protocol package changed the type of 'tagSupport' from bool to an object.
+            // Our version of the LSP protocol package is older and assumes that the type is bool, so deserialization fails.
+            // Since we don't really read this field, just no-op the error until we can update our package references.
+            // https://github.com/dotnet/roslyn/issues/40829 tracks updating this.
+            var settings = new JsonSerializerSettings
+            {
+                Error = (sender, args) =>
+                {
+                    if (object.Equals(args.ErrorContext.Member, "tagSupport") && args.ErrorContext.OriginalObject.GetType() == typeof(PublishDiagnosticsSetting))
```

Comment:
Not just == for the string comparison?

Reply:
member is an object here, so easiest to do it this way